### DISCUSSION TITLE
Write-DbaDbTableData with differents name in csv than sql table

### DISCUSSION THREAD

**Discussion by eusebio72** [dataplat/dbatools#8577] [Q&A] [answered]
Hello,
I use Write-DbaDbTableData to import a csv file into sql. I have no problem if the columns of the file are identical to the sql table, but I don't know how to solve the problem if:

1. the csv file contains more columns than the sql table
2. some column names differ from those of the sql table

I hope you can help me.
Thank you

G.

**Comment by potatoqualitee** [comment, marked as answer]
hey @eusebio72  -- please use the Import-DbaCsv command with either the -Column or -ColumnMap parameter.

**Reply by eusebio72** [reply]
Thanks a lot @potatoqualitee for your reply. Tomorrow I'll try and I tell you

Giorgio

**Reply by eusebio72** [reply]
Thanks @potatoqualitee  your solution is perfec
Regards

Giorgio

**Reply by potatoqualitee** [reply]
Excellent! Glad to help.

**Reply by eusebio72** [reply]
@potatoqualitee  only one other little question. Now it works because column field type are same in csv and sql table. If I need convert a field in onother type while import, how can I do it? For example string in datetime. Thanks

**Reply by potatoqualitee** [reply]
mmm i dont think it does conversions like that on the fly, though you can try. i'd import it into an interim table then do the conversions using SQL Server and T-SQL.

**Reply by eusebio72** [reply]
thanks @potatoqualitee. I done it. I use a temporary table and than I put data with convert in the online table
